webdslstatix/trans/static-semantics/webdsl.stx

```diff
@@ -26,13 +26,20 @@ imports
 signature
 
   sorts
-    TYPE // semantic type
-    BOOL // used as return values of functional constraints
+    TYPE   // semantic type
+    BOOL   // used as return values of functional constraints
+    ORIGIN // dummy data: used to guide scope graph differ
 
   constructors
     TRUE : BOOL
     FALSE : BOOL
 
+    FUNCTION_ORIGIN: list(FormalArg) -> ORIGIN
```

Comment:
IIRC, the `FormalArg` construct also includes the _names_ of the arguments. I think it would be better to filter those, because names are not a part of the signature of a method. So perhaps we should have a `list(Type)` as argument here, and translate all use sites to that form.